nncf/common/pruning/tensor_processor.py

```diff
@@ -74,3 +74,15 @@ def elementwise_mask_propagation(cls, input_masks: List[NNCFTensor]) -> NNCFTens
         :param input_masks: Given input masks.
         :return: Elementwise pruning operation output mask.
         """
+
+    @classmethod
+    @abstractmethod
+    def split(cls, tensor: NNCFTensor, chunks: int, axis: int, output_shapes: List[int] = None) -> List[NNCFTensor]:
```

Comment:
Please remove `output_shapes` param and calculate in in `SymbolicMaskProcessor.Split` by `chunks` param

Reply:
I hope so too, but for the symbolic mask processor, we could not know the result size of the split/chunk operation. The `output_shapes` must be received because the calculation result clearly differs from the chunk size input. In the `SymbolicMaskProcesor`, we could not use chunk or split operations here like `torch.chunk` or `tf.split`, so we need to receive the `output_shapes`.

Reply:
Ok, can we than unify this function across frameworks + SymbolicMaskProcessor by using params
(`tesnor`, `output_shapes`, `axis`)? Looks like `chunks` is just `len(output_shapes)`

Reply:
Yes, right. But some case, we did not put the `output_shapes` as input. So I set `output_shapes` usually None. I want to use `chunks` as much as possible and avoid putting `output_shape`.

Reply:
My concern is that we call function only in one place and always feed all parameters. And depending on realization (PT, TF or Symbolic) different subsets of params is used. `axis` param is needed (possibly) for non symbolic tensors and don't needed for symbolic tensors, but in case of params `chunks` and `output_shapes` - they are interchangeable. And as far as I understand realization just takes one of params which is more compostable for it's realization. This is logical overhead for me

Reply:
I reduced the arguments of the split functions only `output_shapes`. As you said, the mask size is always one dimension, so I reduced the axis and use it as default value (`axis=0`). And remove the `chunks` to be replaced by `len(output_shapes)`.